sdk/sdktypes/value.go

```diff
@@ -30,6 +30,15 @@ func validateValue(m *ValuePB) error {
 		objectField[DictValue]("dict", m.Dict),
 		objectField[StructValue]("struct", m.Struct),
 		objectField[ModuleValue]("module", m.Module),
+		objectField[FunctionValue]("function", m.Function),
```

Comment:
How is this related to the PR? I've expected only `CustomValue` addition here.
This suggests that probably having each type implement a `Validator` interface will be better and more explicit than using protovalidate the reflection magic.